newIDE/app/src/UI/Search/BoxSearchResults.js

```diff
@@ -24,86 +24,126 @@ const styles = {
   grid: { overflowX: 'hidden' },
 };
 
-export const BoxSearchResults = <SearchItem>({
-  searchItems,
-  renderSearchItem,
-  error,
-  onRetry,
-  baseSize,
-  noResultPlaceholder,
-  noScroll,
-}: Props<SearchItem>) => {
-  if (!searchItems) {
-    if (!error) return <PlaceholderLoader />;
-    else {
+export type BoxSearchResultsInterface = {|
+  getScrollPosition: () => number,
+  scrollToPosition: (y: number) => void,
+|};
+
+export const BoxSearchResults = React.forwardRef<
+  // $FlowFixMe The generic type can't pass through.
+  Props<SearchItem>,
+  BoxSearchResultsInterface
+>(
+  (
+    {
+      searchItems,
+      renderSearchItem,
+      error,
+      onRetry,
+      baseSize,
+      noResultPlaceholder,
+      noScroll,
+    }: Props<SearchItem>,
+    ref
+  ) => {
+    const grid = React.useRef<?Grid>(null);
+    React.useImperativeHandle(ref, () => ({
+      getScrollPosition: () => {
+        const gridElement = grid.current;
+        if (!gridElement) return 0;
+
+        // TODO Find a clean way to get the scroll position.
+        // Using the internal state of a component is hacky.
+        // Grid probably doesn't expose the scroll position
+        // because it became irrelevant when the dimensions change.
+        // Though, it's easier to use it and the chance that the Grid is
+        // resized is low.
+        return gridElement.state.scrollTop;
+      },
```

Comment:
This is not great.

Reply:
Did you try to declare a ref that would be `scrollTop` that would be updated on grid scroll?
`<Grid onScroll={(event) => {scrollTop.current = event.currentTarget.scrollTop}} ...>`